iot-hub/Samples/service/RoleBasedAuthenticationSample/RoleBasedAuthenticationSample.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using Microsoft.Azure.Devices;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace RoleBasedAuthenticationSample
+{
+    /// <summary>
+    /// This sample connects to the IoT hub using Azure active directory credentials and sends a cloud-to-device message.
+    /// </summary>
+    public class RoleBasedAuthenticationSample
+    {
+        public async Task RunSampleAsync(ServiceClient client, string deviceId)
+        {
+            Console.WriteLine("Connecting using token credential.");
+            await client.OpenAsync();
+            Console.WriteLine("Success.");
+
+            Console.WriteLine("Sending a cloud-to-device message.");
+            var str = $"Hello, Cloud!";
+            var message = new Message(Encoding.ASCII.GetBytes(str));
+            await client.SendAsync(deviceId, message);
+            message.Dispose();
+            Console.WriteLine("Success.");
+
+            await client.CloseAsync();
```

Comment:
I don't think we like to promote calling `CloseAsync()`. @abhipsaMisra ?

Reply:
Sure but why not?

Reply:
On the device client side, yeah, `CloseAsync()` isn't really useful right now since we cannot guarantee that the transport layer is cleaned up properly. For the service client, since our clients do not come with any recoverable retry logic anyway, we recommend that they dispose and recreate the client when they see an exception that tells then that the transport layer (or amqp links) was disconnected. 

I think what we want here is to avoid people following the open -> close -> open flow, so leaving it out of our samples and instead using dispose to control the lifecycle will help promote that.

Reply:
Nothing will go wrong with the flow that we have now for this sample -> calling close at the end of this sample, but it could possibly encourage people to use it in their applications and run into inconsistent state issues.

Reply:
I can remove it then. Thanks.